### PR TITLE
Create authentication guard

### PR DESCRIPTION
**Pull request by ShahzodK**

### CODE REVIEW

File: src/app/core/guards/auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  canLoad(): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return this.checkToken();
+  }
+
+  public checkToken() {
+    if (localStorage.getItem('authToken')) {
+      return true;
+    }
+    this.router.navigateByUrl('/login');
```
Comment (AdiletMaksatuly) [inline src/app/core/guards/auth.guard.ts:24]: return this.router.parseUrl('login')

that will return UrlTree, as you mentioned in return type of your canActivate and canLoad

File: src/app/core/guards/auth.guard.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthGuard implements CanActivate, CanLoad {
+
+  constructor(private router: Router) {}
+
+  canActivate(): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
```
Comment (AdiletMaksatuly) [inline src/app/core/guards/auth.guard.ts:12]: specify only that types which actually can be returned (boolean | UrlTree)

**Review by AdiletMaksatuly — CHANGES_REQUESTED** [review]

File: src/app/core/guards/auth.guard.ts (hunk 2 above)
Comment (ShahzodK) [reply src/app/core/guards/auth.guard.ts]: Done

File: src/app/core/guards/auth.guard.ts (hunk 1 above)
Comment (ShahzodK) [reply src/app/core/guards/auth.guard.ts]: Done

**Review by AdiletMaksatuly — APPROVED** [review]

**Pull request merged**